### PR TITLE
dev/translation#9 Create API Action to rebuild Multilingual Schema

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
Create an API tool to allow folks to rebuild the multilingual schema

Before
---------------------------------------
Have to use cli tools to rebuild schema

After
----------------------------------------
able to rebuild schema using the API

ping @eileenmcnaughton @mlutfy This isn't the perfect solution at least it should make it easier for folks to get out of trouble hopefully

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@seamuslee001 can we alter an existing test to call this? I could make a case for not having a test here - but it also feels like it might not be that hard if there are existing tests

**Comment by eileenmcnaughton** [comment]
I wonder if we should tie in with system.flush api (that rebuilds triggers if trigger=1)

**Comment by seamuslee001** [comment]
Yeh might be not the worst idea

File: api/v3/System.php [hunk 1]
```diff
@@ -427,3 +427,21 @@ function civicrm_api3_system_createmissinglogtables() {
...
+  $domain = new CRM_Core_DAO_Domain();
+  $domain->find(TRUE);
+
+  if ($domain->locales) {
+    $locales = explode(CRM_Core_DAO::VALUE_SEPARATOR, $domain->locales);
+    CRM_Core_I18n_Schema::rebuildMultilingualSchema($locales);
+    return civicrm_api3_create_success(1);
+  }
+  else {
+    throw new API_Exception('Cannot call rebuild Multilingual scheama on non Multilingual database');
```
Comment (mlutfy) [inline api/v3/System.php]: typo in "scheama"

**Comment by mlutfy** [comment]
I was also initially going to suggest `system.flush multilingual=1`, but seems like we already have a few functions that are not tied to `flush`, ex:

```
function civicrm_api3_system_updatelogtables() {
function civicrm_api3_system_updateindexes() {
function civicrm_api3_system_createmissinglogtables() {
```

whereas we also have `triggers`  on `system.flush`. And to think about it, this is a bit more than a cache flush, so makes sense to keep it separate?

**Comment by eileenmcnaughton** [comment]
Yeah - I'm kinda 6 one way & half a dozen the other on adding to flush vs it's own fn

**Comment by mlutfy** [comment]
@seamuslee001 Can you fix the typo, and we can merge?

**Comment by seamuslee001** [comment]
@mlutfy done

**Comment by eileenmcnaughton** [comment]
Merging this as on balance it helps us deal with support issues & any downside is kinda esoteric

**Pull request merged**